src/file_process.cpp

```diff
@@ -0,0 +1,180 @@
+/** \brief This source file defines functions for Output File Processing.
+*
+* The output file generated by the simulator is processed
+* to produce a more readable format. The data is stored in
+* output file *_proc.txt in a table format as shown below:
+*
+*Time           value  port                component
+*00:00:10:000   20     out                 generator_con
+*00:00:20:000   1      packet_sent_out     sender1
+*00:00:20:000   11     data_out            sender1
+*00:00:23:000   11     out                 subnet1
+*00:00:33:000   1      out                 receiver1
+*00:00:36:000   1      out                 subnet2
+*/
+/*
+ * Peter Bliznyuk-Kvitko
+ * Shubham Agrawal
+ * Carleton University
+*/
+
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <fcntl.h>
+#include <errno.h>
+#include <regex>
+#include <bits/stdc++.h> 
+#include <boost/algorithm/string.hpp>
+
+#include "../include/file_process.hpp"
+
+#define BUF_SIZE 2048
+
+using namespace std;
+
+void output_file_process(char *fin, char *fout) {
+    /**
+     * This function is to check if the read access to the input file is OK
+    */
+    if (access(fin, R_OK)) {
+        std::cout << "The file " << fin << " can not be read from, errno = "
+                  << errno << "\n";
+    }
+    /**
+     * if the access is not ok read the input file into a string and
+     * write the data from the input file string into output file
+    */
+    else {
+        char *file = read_file(fin);
+        write_file(fout, file);
+    }
+}
+
+char *read_file(char *fin) {
+    char read_buf[BUF_SIZE];
+    char *file;
+    int size;
+    int i = 1;
+	
+    int fdi = open(fin, O_RDONLY);
+
+    /**
+     * Check if the file is opened in read only and
```

Comment:
There is a slight mis-match in your way of commenting. Please follow a unique style through the code.

Refer the below images:
![r1](https://user-images.githubusercontent.com/46407088/63236198-da555e00-c20a-11e9-8144-4dce18a26318.jpg)
![r2](https://user-images.githubusercontent.com/46407088/63236206-de817b80-c20a-11e9-98c9-bf9e6de7d7b3.jpg)

Reply:
solved